pytorch_lightning/trainer/trainer.py

```diff
@@ -1383,15 +1383,21 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
             ckpt_path = "best"
 
         if ckpt_path == "best":
-            # if user requests the best checkpoint but we don't have it, error
+            if len(self.checkpoint_callbacks) > 1:
+                raise MisconfigurationException(
+                    f'.{fn}(ckpt_path="best" is not supported with multiple `ModelCheckpoint` callbacks.'
+                    " Please pass in the exact checkpoint path."
+                )
+
             if not self.checkpoint_callback:
                 raise MisconfigurationException(
                     f'`.{fn}(ckpt_path="best")` is set but `ModelCheckpoint` is not configured.'
                 )
+
             if not self.checkpoint_callback.best_model_path:
                 if self.fast_dev_run:
                     raise MisconfigurationException(
-                        f"You cannot execute `.{fn}()` with `fast_dev_run=True` unless you do"
+                        f'You cannot execute `.{fn}(ckpt_path="best")` with `fast_dev_run=True` unless you do'
```

Comment:
I don't get this change. It says

```You cannont execute ckpt_path="best" unless you do ckpt_path=PATH```

But they are not the same thing

Reply:
will update msg